src/shared-components/banner/index.tsx

```diff
@@ -19,7 +19,15 @@ const bannerIconMapping = {
   default: InfoIcon,
 };
 
-const Banner = ({ content, type, onClick }) => {
+export type BannerTypes = 'default' | 'success' | 'error' | 'danger';
+
+type BannerProps = {
+  content: React.ReactNode;
+  type: BannerTypes;
+  onClick: () => void;
+};
+
+const Banner = ({ content, type, onClick }: BannerProps): JSX.Element => {
```

Comment:
It's a little repetitive but we could probably make this more accurate by adding ES6 defaults here in addition to the defaultProps:

```
type BannerProps = {
  content: React.ReactNode;
  type?: BannerTypes;
  onClick?: () => void;
};

const Banner = ({ content, type = 'default', onClick = () => undefined }: BannerProps): JSX.Element => {
```

Otherwise we'll throw errors when using Banner without providing either of those, which doesn't seem like our intent based on the `defaultProps`.

Reply:
Wait, this is important you are right. So my logic of typing props as required to avoid handling undefined state is wrong. I wasn't thinking on this as exported Library. This solves the case for FunctionComponent Ill figure out how to handle class based props

Reply:
Our new convention is Optional Prop but has DefaultProp then mark it as not optional in TS. It works for Functional Components as well so I will leave it without the ES6 default and mark all as TS not optional types

Reply:
tested with this component and works